pkg/plugin/validation.go

```diff
@@ -17,32 +17,64 @@ limitations under the License.
 package plugin
 
 import (
-	"errors"
 	"fmt"
 
 	"github.com/blang/semver"
 
 	"sigs.k8s.io/kubebuilder/pkg/internal/validation"
 )
 
+// errInvalidVersion should be returned when a plugin's version is invalid.
+type errInvalidVersion struct {
```

Comment:
Since we are implementing specific types of error would be good to export them to be used. Also, wdyt about simplified it by: 
```suggestion
type ErrInvalidVersion struct {
```

Reply:
These aren't used anywhere else, yet.